CMakeLists.txt

```diff
@@ -10,21 +16,58 @@ endif(NOT CMAKE_BUILD_TYPE)
 set(CMAKE_CXX_STANDARD 17)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 
-if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU") #GCC flags
+if (${CMAKE_CXX_COMPILER_ID} STREQUAL "GNU") #GCC flags
 	set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wextra -Wno-unused-parameter -Wno-unused-function -Wno-unused-but-set-parameter")
 	set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -flto")
-elseif("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang") #clang flags
+elseif(${CMAKE_CXX_COMPILER_ID} STREQUAL "Clang") #clang flags
 	set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wextra -Wno-unused-parameter -Wno-unused-function")
 	set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -flto")
 else()
 	#Add support for MSVC and ICC if needed
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /DNOMINMAX")
 endif()
 
 list(APPEND CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/cmake")
 
 option(ETALER_ENABLE_OPENCL "Enable the OpenCL backend" OFF)
 
 include_directories(.)
+include_directories(${PROJECT_BINARY_DIR}/Etaler)
+
+############################################################################
+# Find Intel TBB include and libs
+
+# tbbvars.sh (part of binary installation of TBB) uses TBBROOT env var
+if(DEFINED ENV{TBBROOT})
+	set(TBB_DIR "$ENV{TBBROOT}/cmake")
+else()
+	set(TBB_DIR "$ENV{TBB_ROOT_DIR}/cmake")
+endif()
+
+find_package(TBB REQUIRED)
+
+if(NOT ${TBB_FOUND})
+    message(FATAL_ERROR "Could not locate Intel TBB include & libs")
```

Comment:
Indentations in code should be a tab instead of spaces. I should have added a code style guide. My apologies.